File: 190/reverse-bits.py
from functools import reduce


class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        arr = []
        while n > 0:
            bit = n % 2
            arr.append(bit)
            n = (n - bit) / 2
        while len(arr) < 32:
            arr.append(0)
        return reduce(lambda acc, x: acc * 2 + x, arr)

File: 190/reverse-bits.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function (n) {
    let arr = [];
    while (n > 0) {
        let tmp = n % 2;
        arr.push(tmp);
        n = (n - tmp) >> 1;
    }
    while (arr.length < 32) {
        arr.push(0);
    }
    return arr.reduce((acc, x) => acc * 2 + x);
};